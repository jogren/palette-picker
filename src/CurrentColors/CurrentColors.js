import React from 'react';
import Color from '../Color/Color';
import { connect } from 'react-redux';

const CurrentColors = ({ currentPalette, toggleLock }) => {
  let currentColors = currentPalette.map(color => {
    const colorStyle = { backgroundColor: color.hexCode };
    return <Color color={colorStyle} toggleLock={toggleLock} key={color.hexCode} />
  })

  return (
    <section className="current-colors">
      {currentColors}
    </section>
  );
}

const mapStateToProps = ({ currentPalette }) => ({
  currentPalette
})

export default connect(mapStateToProps)(CurrentColors);