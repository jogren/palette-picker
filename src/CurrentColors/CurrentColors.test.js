import React from 'react';
import { shallow } from 'enzyme';
import { CurrentColors, mapStateToProps } from './CurrentColors';

describe('CurrentColors', () => {
  let wrapper;
  const currentPaletteMock = [{
    hexCode: "#25b25d",
    isLocked: false
  }]
  const selectedPalettesMock = [{
    id: 49,
    name: '12345',
    project_id: 13,
    created_at: '2019-10-16T00:35:10.929Z',
    updated_at: '2019-10-16T00:35:10.929Z',
    color1: '#3890af',
    color2: '#1aaf38',
    color3: '#a567c6',
    color4: '#fcf4a1',
    color5: '#7a35c4'
  }]
  const currentProjectsMock = [{
    id: 13,
    name: '123456',
    created_at: '2019-10-13T17:40:38.824Z',
    updated_at: '2019-10-13T17:40:38.824Z'
  }]
  const currentPaletteIdMock = {
    id: 49,
    name: '12345',
    projectId: 13
  }
  const mockState = {
    currentPalette: currentPaletteMock,
    currentProjects: currentProjectsMock,
    selectedPalettes: selectedPalettesMock,
    currentPaletteId: currentPaletteIdMock,
    errorMsg: ""
  }

  beforeEach(() => {
    wrapper = shallow(<CurrentColors
      currentPalette={currentPaletteMock}
    />)
  });

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('mapStateToProps should grab the props it needs', () => {
    const expected = {
      currentPalette: currentPaletteMock
    }

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  });
});