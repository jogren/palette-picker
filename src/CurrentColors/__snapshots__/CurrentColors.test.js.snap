// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentColors should match the snapshot with the data passed through 1`] = `
<section
  className="current-colors"
>
  <Color
    color={
      Object {
        "backgroundColor": "#25b25d",
      }
    }
    isLocked={false}
    key="#25b25d"
  />
</section>
`;
